src/main/java/uk/gov/pay/api/resources/error/ApiErrorResponse.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.pay.api.resources.error;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+import static java.lang.String.format;
+
+@ApiModel(value = "ErrorResponse", description = "An error response")
+@JsonInclude(NON_NULL)
+public class ApiErrorResponse {
+
+    public enum Code {
+
+        TOO_MANY_REQUESTS_ERROR("P0900", "Too many requests");
+
+        private String value;
+        private String format;
+
+        Code(String value, String format) {
+            this.value = value;
+            this.format = format;
+        }
+
+        public String value() {
+            return value;
+        }
+
+        public String getFormat() {
+            return format;
+        }
+    }
+
+    private final Code code;
+    private final String description;
+
+
+    public static ApiErrorResponse anApiErrorResponse(Code code) {
+        return new ApiErrorResponse(code);
+    }
+
+    public static ApiErrorResponse anApiErrorResponse(Code code, Object... parameters) {
+        return new ApiErrorResponse(code, parameters);
+    }
+
+    private ApiErrorResponse(Code code, Object... parameters) {
```

Comment:
this method is never used

Reply:
This is used in RateLimiterFilter. 

https://github.com/alphagov/pay-publicapi/blob/pp_4073_429_response_for_endpoints/src/main/java/uk/gov/pay/api/filter/RateLimiterFilter.java#L70

Reply:
oh oops. looks like i was viewing an older commit locally.